scripts/check_no_panic_in_production.py

```
@@ -0,0 +1,169 @@
+def relpath(path: Path) -> str:
+    return path.relative_to(REPO_ROOT).as_posix()
+
+
+def collect_files() -> list[Path]:
+    files: list[Path] = []
+    for crate in GUARDED_CRATES:
+        crate_src = CRATES_ROOT / crate / "src"
+        if not crate_src.exists():
+            continue
+        for path in sorted(crate_src.rglob("*.rs")):
+            files.append(path)
```

Ruff (0.15.10)

[warning] Use `list` or `list.copy` to create a copy of a list

(PERF402)